mitiq/zne/scaling/layer_scaling.py

```diff
@@ -0,0 +1,74 @@
+# Copyright (C) 2023 Unitary Fund
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+"""Functions for layer-wise unitary folding on supported circuits."""
+from typing import List
+from cirq import Circuit, inverse, Moment
+
+
+def layer_folding(circuit: Circuit, layers_to_fold: List[int]) -> Circuit:
```

Comment:
Do you think we should apply [`noise_scaling_converter`](https://github.com/unitaryfund/mitiq/blob/c849314bc73c2e7b1b71d312639c5e8417ab2e74/mitiq/interface/conversions.py#L221-L223) here as a decorator? It should allow for this function to be used with other frontends.

If we decide to add this, we should test with an additional frontend as well.